app/models/concerns/distanceable.rb

```
@@ -113,6 +174,60 @@ def distance_to(other_point, unit = :km)
+      return 0 if !distance_km.finite?
+
+      result = (distance_km * 1000).to_f / ::DISTANCE_UNITS[unit.to_sym]
+
+      # Final validation
+      return 0 if !result.finite?
+
+      result
+    rescue StandardError => e
+      0
+    end
+  end
```

RuboCop (1.76.1)

[convention] Assignment Branch Condition size for `distance_to_geocoder` is too high. [<9, 37, 22> 43.98/17]

(Metrics/AbcSize)